lib/driver.js

```diff
@@ -410,460 +391,399 @@ DatastaxDriver.prototype.executeCqlOnDriver = function executeCqlOnDriver(pool,
-  _.forEach(params, function (param, index) {
-    if (param && param.hasOwnProperty('value') && param.hasOwnProperty('hint')) {
-      params[index] = param.value;
-      if (param.hint) {
-        if (_.isString(param.hint)) {
-          param.hint = self.dataType.getByName(param.hint);
-        }
-        hints[index] = param.hint;
+  async _timeQuery(queryRequestId, cqlQuery, dataParams, options, act) {
+    try {
+      const metrics = this.metrics;
+      const captureMetrics = !!(metrics && options.queryName);
+      const start = captureMetrics && process.hrtime();
+      this.emit('queryStarted', queryRequestId, cqlQuery, dataParams, options.consistency, options);
+      const result = await act(options);
+      this.emit('queryCompleted', queryRequestId);
+      if (captureMetrics) {
+        var duration = process.hrtime(start);
+        duration = (duration[0] * 1e3) + (duration[1] / 1e6);
+        metrics.measurement('query.' + options.queryName, duration, 'ms');
       }
-      if (param.isRoutingKey) {
-        routingIndexes.push(index);
+      return result;
+    } catch (err) {
+      this.emit('queryFailed', queryRequestId);
+      throw err;
+    }
+  }
+
+  _prepareQueryArgs(pool, cqlStatement, params, consistency, options) {
+    var self = this;
+    var execOptions = _.assign({
+      prepare: !!options.executeAsPrepared,
+      consistency: consistency
+    }, options);
+
+    var hints = [];
+    var routingIndexes = [];
+    _.forEach(params, function (param, index) {
+      if (param && param.hasOwnProperty('value') && param.hasOwnProperty('hint')) {
+        params[index] = param.value;
+        if (param.hint) {
+          if (_.isString(param.hint)) {
+            param.hint = self.dataType.getByName(param.hint);
+          }
+          hints[index] = param.hint;
+        }
+        if (param.isRoutingKey) {
+          routingIndexes.push(index);
+        }
       }
+    });
+    if (hints.length) {
+      execOptions.hints = hints;
     }
-  });
-  if (hints.length) {
-    execOptions.hints = hints;
-  }
-  if (routingIndexes.length) {
-    execOptions.routingIndexes = routingIndexes;
+    if (routingIndexes.length) {
+      execOptions.routingIndexes = routingIndexes;
+    }
+    return { cqlStatement: cqlStatement, params: params, execOptions: execOptions };
   }
-  return { cqlStatement: cqlStatement, params: params, execOptions: execOptions };
-};
 
-DatastaxDriver.prototype.getDriverDataType = function getDriverDataType(type) {
-  // convert 'objectAscii' type to 'ascii', etc.
-  var prop;
-  for (prop in this.dataType) {
-    if (this.dataType.hasOwnProperty(prop) && this.dataType[prop] === type) {
-      // check if property starts with Object
-      if (prop.length > 7 && prop.substring(0, 6) === 'object') {
-        prop = prop.substring(6, 7).toLowerCase() + prop.substring(7);
-        return this.dataType[prop];
+  _getDriverDataType(type) {
+    // convert 'objectAscii' type to 'ascii', etc.
```

Comment:
This dataType conversion is also related to the old drivers.

Reply:
Nevermind, this is Priam converting Objects to JSON and storing in either `ascii` or `text` fields.